src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -514,74 +586,27 @@ std::string ModifierManager::get_flat_identifier(
 	return StringUtils::append_string_views(complex_modifier_identifier, " ", variant_identifier);
 }
 
-// We use std::string const& identifier instead of std::string_view identifier as the map [] lookup operator only accepts
-// strings. In order to use string_views we need to use the find method but that returns an iterator with a const references
-// to the key and value (in order to prevent modification of the key), so it's simplest to just use [] with a string.
-bool ModifierManager::register_modifier_effect_variants(
-	std::string const& identifier, ModifierEffect const* effect, std::vector<Modifier::modifier_type_t> const& types
-) {
-	if (identifier.empty()) {
-		Logger::error("Invalid modifier effect variants identifier - empty!");
-		return false;
-	}
-
-	if (effect == nullptr) {
-		Logger::error("Invalid modifier effect variants effect for \"", identifier, "\" - nullptr!");
-		return false;
-	}
-
-	if (types.empty()) {
-		Logger::error("Invalid modifier effect variants types for \"", identifier, "\" - empty!");
-		return false;
-	}
-
-	effect_variant_map_t& variant_map = modifier_effect_variants[identifier];
-
-	bool ret = true;
-
-	for (const Modifier::modifier_type_t type : types) {
-		ModifierEffect const*& variant_effect = variant_map[type];
-
-		if (variant_effect != nullptr) {
-			Logger::error(
-				"Duplicate modifier effect variant for \"", identifier, "\" with type \"",
-				Modifier::modifier_type_to_string(type), "\" - already registered as \"",
-				variant_effect->get_identifier(), "\", setting to \"", effect->get_identifier(), "\""
-			);
-			ret = false;
-		}
-
-		variant_effect = effect;
-	}
-
-	return ret;
-}
-
-bool ModifierManager::add_event_modifier(std::string_view identifier, ModifierValue&& values, IconModifier::icon_t icon) {
-	using enum Modifier::modifier_type_t;
+bool ModifierManager::add_event_modifier(std::string_view identifier, ModifierValue&& values, IconModifier::icon_t icon, Modifier::modifier_origin_t origin) {
 
 	if (identifier.empty()) {
 		Logger::error("Invalid event modifier effect identifier - empty!");
 		return false;
 	}
 
 	return event_modifiers.add_item(
-		{ identifier, std::move(values), EVENT, icon }, duplicate_warning_callback
+		{ identifier, std::move(values), origin, icon }, duplicate_warning_callback
 	);
 }
 
-bool ModifierManager::load_event_modifiers(ast::NodeCPtr root) {
+bool ModifierManager::load_event_modifiers(ast::NodeCPtr const& root) {
```

Comment:
These shouldn't be `const&`, `ast::NodeCPtr` is an alias of a pointer type so a reference to it would be equivalent to a pointer to a pointer when we could just pass the first pointer by value.

Reply:
Normal const doesn't do anything bad but isn't really necessary for function args